library/src/main/java/com/schibsted/spain/barista/internal/matcher/SeekBarAssertions.kt

```diff
@@ -0,0 +1,60 @@
+package com.schibsted.spain.barista.internal.matcher
+
+import android.support.test.espresso.AmbiguousViewMatcherException
+import android.support.test.espresso.Espresso
+import android.support.test.espresso.NoMatchingViewException
+import android.support.test.espresso.ViewAssertion
+import android.support.test.espresso.assertion.ViewAssertions
+import android.support.test.espresso.matcher.ViewMatchers
+import android.view.View
+import android.widget.SeekBar
+import com.schibsted.spain.barista.internal.failurehandler.SpyFailureHandler
+import com.schibsted.spain.barista.internal.failurehandler.description
+import org.hamcrest.Matcher
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.`is`
+
+object SeekBarAssertions {
+  @JvmStatic
+  fun assertProgress(expectedProgress: Int): ViewAssertion {
```

Comment:
Have you consider implementing this with a BoundedMatcher instead of a ViewAssertion? 
Matchers have a higher-level api, and automatically handle thing like casting the view, throwing exceptions and creating nice error messages.

You can create them anonymously, or with their own class like for example:
```kotlin
class ProgressMatcher(private val expectedProgress: Int): BoundedMatcher<View, ProgressBar>(ProgressBar::class.java) {

    override fun matchesSafely(progressBar: ProgressBar) = progressBar.progress == expectedProgress

    override fun describeTo(description: Description) {
      description.appendText("with progress $expectedProgress")
    }

}
```

The difference in usage is that with a ViewAssertion you do:
```
onView(...).check(myAssertion)
```
but with a matcher you do the more common:
```
onView(...).check(matches(myMatcher))
```

Reply:
I will take a look, I based the current implementation in `RecyclerViewItemCountAssertion` where it inherits from `ViewAssertion` but `BoundedMatcher` seems to be cleaner.

Reply:
Ouch you're right, I forgot we had that one as a ViewAssertion. It's from the days we didn't know that much about Espresso, sorry :P
I will try and migrate to a matcher for consistency.